crates/oxc_linter/src/snapshots/prefer_prototype_methods.snap

```diff
@@ -0,0 +1,143 @@
+---
+source: crates/oxc_linter/src/tester.rs
+expression: prefer_prototype_methods
+---
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using `Array.prototype.push`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
+ 1 │ const foo = [].push.apply(bar, elements);
+   ·             ───────
+   ╰────
+
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using `Array.prototype.slice`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
+ 1 │ const foo = [].slice.call(bar);
+   ·             ────────
+   ╰────
+
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using `Object.prototype.toString`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
+ 1 │ const foo = {}.toString.call(bar);
+   ·             ───────────
+   ╰────
+
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using `Object.prototype.hasOwnProperty`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
+ 1 │ const foo = {}.hasOwnProperty.call(bar, "property");
+   ·             ─────────────────
+   ╰────
+
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using `Object.prototype.propertyIsEnumerable`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
+ 1 │ const foo = {}.propertyIsEnumerable.call(bar, "property");
+   ·             ───────────────────────
+   ╰────
+
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using `Array.prototype.forEach`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
+ 1 │ [].forEach.call(foo, () => {})
+   · ──────────
+   ╰────
+
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using `Array.prototype.push`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
+ 1 │ const push = [].push.bind(foo)
+   ·              ───────
+   ╰────
+
+  ⚠ eslint-plugin-unicorn(prefer-prototype-methods): Prefer using method from `Array.prototype`.
+   ╭─[prefer_prototype_methods.tsx:1:1]
```

Comment:
It looks like there should be `Array.prototype.method`

Reply:
For this test case, the error message generated by eslint-plugin-unicorn is also "Prefer using method from `Array.prototype`." (See [the snapshot file](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/82d3de639ef3041034c4e8c5973ac7ff39b5aaed/test/snapshots/prefer-prototype-methods.mjs.md?plain=1#L128-L133) in the eslint-plugin-unicorn repo.)

I compared my snapshot file with theirs. There are only two differences in the generated error message. (The fixed code for each case below is correct. Only the error message is different from that of eslint-plugin-unicorn.)

## Case 1

Code: `const method = "realMethodName";const foo = [][method].call(foo)`
Error message generated by eslint-plugin-unicorn: ``Prefer using method from `Array.prototype`.``
Error message generated by my code: ``Prefer using method from `Array.prototype`.``
Analysis: When we have a utility function similar to `getPropertyName` in @eslint-community/eslint-utils, the error message for this case can be easily fixed.

## Case 2

Code: `[][Symbol.iterator].call(foo)`
Error message generated by eslint-plugin-unicorn: ``Prefer using `Array.prototype.Symbol(Symbol.iterator)`.``
Error message generated by my code: ``Prefer using method from `Array.prototype`.``
Analysis: The error message generated by eslint-plugin-unicorn does not look correct to me because `Array.prototype.Symbol(Symbol.iterator)` is not valid JavaScript code, so I am unsure about the expected error message for this case.

What do you think about these two cases?

(Btw, to reply to your comment above, I actually generated the rule file using the `just` command you mentioned, but I did some manual editing to the generated test cases like removing unnecessary `r#` prefixes, so it might look like I manually created the rule file, but thanks for the pointer.)

Reply:
Thanks for taking the time to investigate here.

I think we can keep the current error message. Looks better than unicorn